pkg/dispatcher/dispatcher/retry.go

```diff
@@ -58,13 +60,21 @@ func (d *Dispatcher) Dispatch(event *cloudevents.Event, subscription Subscriptio
 	var deadLetterURL *url.URL
 	if subscription.Delivery != nil && subscription.Delivery.DeadLetterSink != nil && !subscription.Delivery.DeadLetterSink.URI.IsEmpty() {
```

Comment:
you may want to use `pkg/resolver/URIResolver` to support object reference

Reply:
👍  Not a part of original scope (existing code borrowed from eventing-contrib) but would be great to harden - thanks for suggesting!